x/fungible/keeper/msg_server_update_gateway_contract.go

```diff
@@ -0,0 +1,53 @@
+package keeper
+
+import (
+	"context"
+
+	cosmoserrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	authoritytypes "github.com/zeta-chain/zetacore/x/authority/types"
+	"github.com/zeta-chain/zetacore/x/fungible/types"
+)
+
+// UpdateGatewayContract updates the zevm gateway contract used by the ZetaChain protocol to read inbounds and process outbounds
+func (k msgServer) UpdateGatewayContract(
+	goCtx context.Context,
+	msg *types.MsgUpdateGatewayContract,
+) (*types.MsgUpdateGatewayContractResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	err := k.GetAuthorityKeeper().CheckAuthorization(ctx, msg)
+	if err != nil {
+		return nil, cosmoserrors.Wrap(authoritytypes.ErrUnauthorized, err.Error())
```

Comment:
Question: I see this a lot of similar patterns like this `cosmoserrors.Wrap(authoritytypes.ErrUnauthorized, err.Error())`. Why do we wrap sentinelError with err.Error()? Why not return original errors if they are already a cosmoserror inside this keeper?

e.g. 
```suggestion
		return nil, cosmoserrors.Wrap(err, "unauthorized")
```

Reply:
We mainly use this pattern to assert with `authoritytypes.ErrUnauthorized` in unit test, but this might not be the best pattern. We can consider change it and create an issue for it

Reply:
https://github.com/zeta-chain/node/issues/2593